### PR TITLE
Better distance_2 docs

### PR DESCRIPTION
**Pull request by urschrei**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `rstar/CHANGELOG.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: rstar/src/object.rs [hunk 1]
```diff
@@ -145,7 +145,12 @@ pub trait RTreeObject {
 /// assert!(circle.contains_point(&[1.0, 0.0]));
 /// ```
 pub trait PointDistance: RTreeObject {
-    /// Returns the squared euclidean distance between an object to a point.
+    /// Returns the squared distance between an object to a point.
```
Comment (michaelkirk) [inline rstar/src/object.rs:148]:
```suggestion
    /// Returns the squared distance between an object and a point.
```

Or alternatively: 

```
 /// Returns the squared distance from an object to a point.
```

(I know this wasn't your text originally 😉)

**Review by michaelkirk — APPROVED** [review]

**Comment by adamreichold** [comment]
I think the `Envelope` trait has the same issue for its `distance_2` method.

**Comment by urschrei** [comment]
@adamreichold Good catch. Is it worth cautioning implementers that the two trait method implementations should use the same metric? I.e. can you imagine a situation in which they might be different?

**Comment by adamreichold** [comment]
> @adamreichold Good catch. Is it worth cautioning implementers that the two trait method implementations should use the same metric? I.e. can you imagine a situation in which they might be different?

From how it is used in the nearest neighbour search, I think these two implementations of the underlying must definitely agree as the distance from envelopes to points is used as lower bound for the distance between objects inside these envelopes to these points.

(From a semantics point of view, it would probably be nicer if the envelope type would implement `PointDistance` just as the object type does to stress this equivalence, but I guess this is not worth it to change retroactively.)

**Review by adamreichold — APPROVED** [review]

**Pull request merged**